contracts/core/extensions/CoreIssuanceOrder.sol

```diff
@@ -272,4 +275,57 @@ contract CoreIssuanceOrder is
             INVALID_QUANTITY
         );
     }
+
+    function settleOrder(
+        OrderLibrary.IssuanceOrder _order,
+        uint _fillQuantity,
+        bytes _orderData
+    )
+        private
+    {
+        uint[] requiredBalances;
+
+        // Calculate amount of maker token required
+        // Look into rounding errors
+        uint requiredMakerToken = _fillQuantity.mul(_order.makerTokenAmount).div(_order.quantity);
+
+        // Get Set Token information
+        address[] memory components = ISetToken(_order.setAddress).getComponents();
+        uint[] memory componentUnits = ISetToken(_order.setAddress).getUnits();
+
+        // Calculate amount of component tokens required to issue
+        for (uint16 i = 0; i < components.length; i++) {
```

Comment:
I wonder if this is necessary. Issue will fail automatically if the correct component amounts are not either in the vault or the users account